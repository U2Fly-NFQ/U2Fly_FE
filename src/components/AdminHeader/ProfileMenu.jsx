import React from 'react'
import { Menu } from 'antd'
import { LogoutOutlined } from '@ant-design/icons'
import { getLsObj } from '../../utils/localStorage'
import { useNavigate } from 'react-router-dom'

function ProfileMenu() {
  const user = getLsObj('user')
  const navigate = useNavigate()

  return (
    <Menu
      items={[
        {
          key: '1',
          label: user.username,
        },
        {
          key: '2',
          icon: <LogoutOutlined />,
          label: 'Logout',
          onClick: () => {
            localStorage.removeItem('user')
            localStorage.removeItem('token')
            navigate('/')
          },
        },
      ]}
    />
  )
}

export default ProfileMenu
