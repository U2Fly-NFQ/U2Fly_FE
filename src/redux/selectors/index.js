export * from './auth'
export * from './bookingFlightSelector'
export * from './ticket'
