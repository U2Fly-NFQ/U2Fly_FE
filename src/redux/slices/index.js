export * from './searchSlice'
export * from './filterSlice'
export * from './flightSlice'
