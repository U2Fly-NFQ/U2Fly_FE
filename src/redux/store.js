import { configureStore } from '@reduxjs/toolkit'
import bookingFlightsSlice from './slices/bookingFlightsSlice'

// import loginSlice from './slices/loginSlice'
import flightSlice from './slices/flightSlice'
import airportSlice from './slices/airportSlice'

const store = configureStore({
  reducer: {
    // login: loginSlice.reducer,
    bookingFlight: bookingFlightsSlice.reducer,
    flights: flightSlice.reducer,
    airports: airportSlice.reducer,
  },
})

export default store
