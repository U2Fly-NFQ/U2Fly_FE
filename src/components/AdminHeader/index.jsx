import React, { useState } from 'react'
import { Avatar, Dropdown, Layout, Input, Row, Col } from 'antd'
import { UserOutlined } from '@ant-design/icons'
import ProfileMenu from './ProfileMenu'

const { Header } = Layout
const { Search } = Input

function AdminHeader() {
  //initiation
  const [search, setSearch] = useState('')

  return (
    <Header className="admin-header">
      <Row className="admin-header-content">
        <Col span={12} className="admin-header-content-left">
          <Search
            placeholder="Search in app"
            enterButton="Search"
            value={search}
          />
        </Col>
        <Col span={12} className="admin-header-content-right">
          <Dropdown className="profileMenu" overlay={<ProfileMenu />}>
            <Avatar size="large" icon={<UserOutlined />} />
          </Dropdown>
        </Col>
      </Row>
    </Header>
  )
}

export default AdminHeader
