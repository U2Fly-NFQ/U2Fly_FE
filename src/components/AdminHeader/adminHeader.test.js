import { I18nextProvider } from 'react-i18next'
import i18n from '../../translations'
import React, { Suspense } from 'react'
import { store } from '../../redux/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'

import AdminHeader from './'
import { ConfigProvider } from 'antd'
import renderer from 'react-test-renderer'

describe('Admin header Layout', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <I18nextProvider i18n={i18n}>
        <Provider store={store}>
          <ConfigProvider>
            <Router>
              <Suspense>
                <AdminHeader />
              </Suspense>
            </Router>
          </ConfigProvider>
        </Provider>
      </I18nextProvider>
    )

    expect(tree).toMatchSnapshot()
  })
})
