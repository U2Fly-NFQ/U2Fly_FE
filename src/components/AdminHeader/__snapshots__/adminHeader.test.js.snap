// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin header Layout renders correctly 1`] = `
<div
  className="ant-row admin-header-content"
  style={Object {}}
>
  <div
    className="ant-col admin-header-content-left ant-col-xs-0 ant-col-sm-12"
    style={Object {}}
  />
  <div
    className="ant-col admin-header-content-right ant-col-xs-24 ant-col-sm-12"
    style={Object {}}
  >
    <div
      className="ant-space ant-space-horizontal ant-space-align-center ant-dropdown-trigger"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "white",
        }
      }
    >
      <div
        className="ant-space-item"
        style={
          Object {
            "marginRight": 8,
          }
        }
      >
        career@nfq.asia
      </div>
      <div
        className="ant-space-item"
        style={Object {}}
      >
        <span
          aria-label="down"
          className="anticon anticon-down"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
